### PR TITLE
Spanish translation for section async

### PR DESCRIPTION
**Pull request by alvaropernas**
Hi, 
Here is my first approach.
Appreciate any comments, thks!!

### CODE REVIEW

File: docs/es/docs/async.md [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+Cuando hay una operación que requerirá esperar antes de dar los resultados y tiene soporte para estas nuevas características de Python, puede codificarlo como:
+
+```Python
+burgers = await get_burgers(2)
+```
+
+
+La clave aquí es la 'espera'. Le dice a Python que tiene que esperar a que `get_burgers (2)` termine de hacer lo suyo antes de almacenar los resultados en `hamburguesas`. Con eso, Python sabrá que puede ir y hacer otra cosa mientras tanto (como recibir otra solicitud).
+
+Para que `await` funcione, tiene que estar dentro de una función que admita esta asincronía. Para hacer eso, simplemente lo declara con `async def`:
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:279]:
```suggestion
Para que `await` funcione, tiene que estar dentro de una función que admita esta asincronía. Para hacer eso, simplemente lo declaras con `async def`:
```

File: docs/es/docs/async.md [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+...instead of `def`:
+
+```Python hl_lines="2"
+# This is not asynchronous
+def get_sequential_burgers(number: int):
+    # Do some sequential stuff to create the burgers
+    return burgers
+```
+
+Con `async def`, Python sabe que, dentro de esa función, debe tener en cuenta las expresiones` wait ', y que puede "pausar" la ejecución de esa función e ir a hacer otra cosa antes de regresar.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:296]:
```suggestion
Con `async def`, Python sabe que, dentro de esa función, debe tener en cuenta las expresiones `wait` y que puede "pausar" la ejecución de esa función e ir a hacer otra cosa antes de regresar.
```

File: docs/es/docs/async.md [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+
+Cuando desee llamar a una función `async def`, debe" esperarla ". Entonces, esto no funcionará:
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:298]:
```suggestion
Cuando desees llamar a una función `async def`, debes" esperarla ". Entonces, esto no funcionará:
```

File: docs/es/docs/async.md [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+
+```Python
+# Esto no funcionará, porque get_burgers se definió con: async def
+hamburguesas = get_burgers (2)
+```
+
+---
+
+Por lo tanto, si está utilizando una biblioteca que le dice que puede llamarla con `await`, debe crear las *funciones de enrutado* que la usan con` async def`, como en:
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:307]:
```suggestion
Por lo tanto, si estás utilizando una biblioteca que te dice que puedes llamarla con `await`, debes crear las *path operation functions* que la usan con `async def`, como en:
```

File: docs/es/docs/async.md [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+async def read_burgers():
+    burgers = await get_burgers(2)
+    return burgers
+```
+
+### Más detalles técnicos
+
+Es posible que haya notado que 'esperar' solo se puede usar dentro de las funciones definidas con 'async def'.
+
+Pero al mismo tiempo, las funciones definidas con `async def` deben ser "esperadas". Por lo tanto, las funciones con `async def` solo se pueden invocar dentro de las funciones definidas con` async def` también.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:320]:
```suggestion
Pero al mismo tiempo, las funciones definidas con `async def` deben ser "esperadas". Por lo tanto, las funciones con `async def` solo se pueden invocar dentro de las funciones definidas con `async def` también.
```

File: docs/es/docs/async.md [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+```Python hl_lines="2 3"
+@app.get('/burgers')
+async def read_burgers():
+    burgers = await get_burgers(2)
+    return burgers
+```
+
+### Más detalles técnicos
+
+Es posible que haya notado que 'esperar' solo se puede usar dentro de las funciones definidas con 'async def'.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:318]:
```suggestion
Es posible que hayas notado que `await` solo se puede usar dentro de las funciones definidas con `async def`.
```

File: docs/es/docs/async.md [hunk 7]
```diff
@@ -0,0 +1,397 @@
...
+
+Pero al mismo tiempo, las funciones definidas con `async def` deben ser "esperadas". Por lo tanto, las funciones con `async def` solo se pueden invocar dentro de las funciones definidas con` async def` también.
+
+Entonces, relacionado con la paradoja del huevo y la gallina, ¿cómo se llama a la primera función `async`?
+
+Si está trabajando con **FastAPI** no tiene que preocuparse por eso, porque esa "primera" función será su *función de enrutados*, y FastAPI sabrá cómo hacer lo pertinente.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:324]:
```suggestion
Si estás trabajando con **FastAPI** no tienes que preocuparte por eso, porque esa "primera" función será tu *path operation function*, y FastAPI sabrá cómo hacer lo pertinente.
```

File: docs/es/docs/async.md [hunk 8]
```diff
@@ -0,0 +1,397 @@
...
+
+En el caso de que desee usar `async` /` await` sin FastAPI, <a href="https://docs.python.org/3/library/asyncio-task.html#coroutine" class="external-link" target="_blank">check the official Python docs</a>.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:326]:
```suggestion
En el caso de que desees usar `async` / `await` sin FastAPI, <a href="https://docs.python.org/3/library/asyncio-task.html#coroutine" class="external-link" target="_blank">revisa la documentación oficial de Python</a>.
```

File: docs/es/docs/async.md [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+
+### Otras formas de código asíncrono
+
+Este estilo de usar `async` y` await` es relativamente nuevo en el lenguaje.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:330]:
```suggestion
Este estilo de usar `async` y `await` es relativamente nuevo en el lenguaje.
```

File: docs/es/docs/async.md [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+
+Pero hace que trabajar con código asíncrono sea mucho más fácil.
+
+Esta misma sintaxis (o casi idéntica) también se incluyó recientemente en las versiones modernas de JavaScript (en Browser y NodeJS).
+
+Pero antes de eso, manejar código asíncrono era bastante más complejo y difícil.
+
+En versiones anteriores de Python, podría haber utilizado hilos o <a href="http://www.gevent.org/" class="external-link" target="_blank">Gevent</a>. Pero el código es mucho más complejo de entender, depurar y desarrollar.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:338]:
```suggestion
En versiones anteriores de Python, podrías haber utilizado hilos o <a href="http://www.gevent.org/" class="external-link" target="_blank">Gevent</a>. Pero el código es mucho más complejo de entender, depurar y desarrollar.
```

File: docs/es/docs/async.md [hunk 11]
```diff
@@ -0,0 +1,397 @@
...
+
+En versiones anteriores de NodeJS / Browser JavaScript, habría utilizado "callbacks". Lo que conduce a <a href="http://callbackhell.com/" class="external-link" target="_blank"> callback hell </a>.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:340]:
```suggestion
En versiones anteriores de NodeJS / Browser JavaScript, habrías utilizado "callbacks". Lo que conduce a <a href="http://callbackhell.com/" class="external-link" target="_blank">callback hell</a>.
```

File: docs/es/docs/async.md [hunk 12]
```diff
@@ -0,0 +1,397 @@
...
+
+## Corutinas
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:342]:
```suggestion
## Coroutines
```

File: docs/es/docs/async.md [hunk 13]
```diff
@@ -0,0 +1,397 @@
...
+
+**Coroutine** es el término guay para la cosa devuelta por una función `async def`. Python sabe que es algo así como una función que puede iniciar y que terminará en algún momento, pero que también podría pausarse internamente, siempre que haya una `await` dentro de ella.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:344]:
```suggestion
**Coroutine** es un término sofisticado para referirse a la cosa devuelta por una función `async def`. Python sabe que es algo así como una función que puede iniciar y que terminará en algún momento, pero que también podría pausarse internamente, siempre que haya una `await` dentro de ella.
```

File: docs/es/docs/async.md [hunk 14]
```diff
@@ -0,0 +1,397 @@
...
+
+Pero toda esta funcionalidad de usar código asincrónico con `async` y` await` se resume muchas veces como el uso de "corutinas". Es comparable a la característica principal de Go, las "Goroutines".
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:346]:
```suggestion
Pero toda esta funcionalidad de usar código asincrónico con `async` y `await` se resume muchas veces como el uso de "coroutines". Es comparable a la característica principal de Go, las "Goroutines".
```

File: docs/es/docs/async.md [hunk 15]
```diff
@@ -0,0 +1,397 @@
...
+
+## Conclusión
+
+Veamos la misma frase de arriba:
+
+> Las versiones modernas de Python tienen soporte para **"código asíncrono"** usando algo llamado **"corutinas"**, con la sintaxis **`async` y` await`**.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:352]:
```suggestion
> Las versiones modernas de Python tienen soporte para **"código asíncrono"** usando algo llamado **"coroutines"**, con la sintaxis **`async` y `await`**.
```

File: docs/es/docs/async.md [hunk 16]
```diff
@@ -0,0 +1,397 @@
...
+Todo eso es lo que impulsa FastAPI (a través de Starlette) y lo que lo hace tener un rendimiento tan impresionante.
+
+## Detalles muy técnicos
+
+!!! advertencia
+    Probablemente puedas saltarte esto.
+
+    Estos son detalles muy técnicos de cómo **FastAPI** funciona a muy bajo nivel.
+    
+    Si tienes bastante conocimiento técnico (corutinas, hilos, bloqueos, etc.) y tienes curiosidad acerca de cómo FastAPI gestiona `async def` vs normal `def`, continúa.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:365]:
```suggestion
    Si tienes bastante conocimiento técnico (coroutines, threads, bloqueos, etc.) y tienes curiosidad acerca de cómo FastAPI gestiona `async def` vs `def` normal, continúa.
```

File: docs/es/docs/async.md [hunk 17]
```diff
@@ -0,0 +1,397 @@
...
+
+### Funciones de enrutado
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:367]:
```suggestion
### Path operation functions 
```

File: docs/es/docs/async.md [hunk 18]
```diff
@@ -0,0 +1,397 @@
...
+!!! advertencia
+    Probablemente puedas saltarte esto.
+
+    Estos son detalles muy técnicos de cómo **FastAPI** funciona a muy bajo nivel.
+    
+    Si tienes bastante conocimiento técnico (corutinas, hilos, bloqueos, etc.) y tienes curiosidad acerca de cómo FastAPI gestiona `async def` vs normal `def`, continúa.
+
+### Funciones de enrutado 
+
+Cuando declara una *función de enrutado* con `def` normal en lugar de `async def`, se ejecuta en un conjunto de subprocesos externo que luego  espera, en lugar de ser llamado directamente (ya que bloquearía el servidor).
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:369]:
```suggestion
Cuando declaras una *path operation function* con `def` normal en lugar de `async def`, se ejecuta en un threadpool externo que luego espera, en lugar de ser llamado directamente (ya que bloquearía el servidor).
```

File: docs/es/docs/async.md [hunk 19]
```diff
@@ -0,0 +1,397 @@
...
+
+Si vienes de otro framework asíncrono que no funciona de la manera descrita anteriormente y estás acostumbrado a definir funciones de enrutado *del tipo sólo cálculo* con `def` simple para una pequeña ganancia de rendimiento (aproximadamente 100 nanosegundos), tenga en cuenta que en **FastAPI** el efecto sería bastante opuesto. En estos casos, es mejor usar `async def` a menos que sus *funciones de enrutado* usen un código que realice el bloqueo <abbr title="Input/Output: disk reading or writing, network communications.">IO</abbr>.
+
+Aún así, en ambas situaciones, es probable que **FastAPI** sea [aún más rápido](/#rendimiento) {.Internal-link target=_blank} que (o al menos comparable) a su framework anterior.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:373]:
```suggestion
Aún así, en ambas situaciones, es probable que **FastAPI** sea [aún más rápido](/#rendimiento){.Internal-link target=_blank} que (o al menos comparable) a tu framework anterior.
```

File: docs/es/docs/async.md [hunk 20]
```diff
@@ -0,0 +1,397 @@
...
+
+    Estos son detalles muy técnicos de cómo **FastAPI** funciona a muy bajo nivel.
+    
+    Si tienes bastante conocimiento técnico (corutinas, hilos, bloqueos, etc.) y tienes curiosidad acerca de cómo FastAPI gestiona `async def` vs normal `def`, continúa.
+
+### Funciones de enrutado 
+
+Cuando declara una *función de enrutado* con `def` normal en lugar de `async def`, se ejecuta en un conjunto de subprocesos externo que luego  espera, en lugar de ser llamado directamente (ya que bloquearía el servidor).
+
+Si vienes de otro framework asíncrono que no funciona de la manera descrita anteriormente y estás acostumbrado a definir funciones de enrutado *del tipo sólo cálculo* con `def` simple para una pequeña ganancia de rendimiento (aproximadamente 100 nanosegundos), tenga en cuenta que en **FastAPI** el efecto sería bastante opuesto. En estos casos, es mejor usar `async def` a menos que sus *funciones de enrutado* usen un código que realice el bloqueo <abbr title="Input/Output: disk reading or writing, network communications.">IO</abbr>.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:371]:
```suggestion
Si vienes de otro framework asíncrono que no funciona de la manera descrita anteriormente y estás acostumbrado a definir *path operation functions* del tipo sólo cálculo con `def` simple para una pequeña ganancia de rendimiento (aproximadamente 100 nanosegundos), ten en cuenta que en **FastAPI** el efecto sería bastante opuesto. En estos casos, es mejor usar `async def` a menos que tus *path operation functions* usen un código que realice el bloqueo <abbr title="Input/Output: disk reading or writing, network communications.">IO</abbr>.
```

File: docs/es/docs/async.md [hunk 21]
```diff
@@ -0,0 +1,397 @@
...
+
+Aún así, en ambas situaciones, es probable que **FastAPI** sea [aún más rápido](/#rendimiento) {.Internal-link target=_blank} que (o al menos comparable) a su framework anterior.
+
+### Dependencias
+
+Lo mismo se aplica para las dependencias. Si una dependencia es una función estándar `def` en lugar de` async def`, se ejecuta en el conjunto de hilos externo.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:377]:
```suggestion
Lo mismo se aplica para las dependencias. Si una dependencia es una función estándar `def` en lugar de `async def`, se ejecuta en el threadpool externo.
```

File: docs/es/docs/async.md [hunk 22]
```diff
@@ -0,0 +1,397 @@
...
+
+### Subdependencias
+
+Puede tener múltiples dependencias y subdependencias que se requieren entre sí (como parámetros de las definiciones de función), algunas de ellas pueden crearse con `async def` y otras con` def` normal. Seguiría funcionando correctamente, y los creados con `def` normal se llamarían en un hilo externo en lugar de ser " awaited ".
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:381]:
```suggestion
Puedes tener múltiples dependencias y subdependencias que se requieren entre sí (como parámetros de las definiciones de función), algunas de ellas pueden crearse con `async def` y otras con `def` normal. Seguiría funcionando correctamente, y los creados con `def` normal se llamarían en un thread externo en lugar de ser "awaited".
```

File: docs/es/docs/async.md [hunk 23]
```diff
@@ -0,0 +1,397 @@
...
+
+### Otras "utility functions"
+
+Cualquier otra función de utilidad que llame directamente se puede crear con `def` o` async def` normales y FastAPI no afectará la manera en que la llama.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:385]:
```suggestion
Cualquier otra función de utilidad que llame directamente se puede crear con `def` o `async def` normales y FastAPI no afectará la manera en que la llamas.
```

File: docs/es/docs/async.md [hunk 24]
```diff
@@ -0,0 +1,397 @@
...
+
+Esto choca con las funciones que FastAPI le solicita: *funciones de enrutado* y dependencias.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:387]:
```suggestion
Esto contrasta con las funciones que FastAPI llama por ti: *path operation functions* y dependencias.
```

File: docs/es/docs/async.md [hunk 25]
```diff
@@ -0,0 +1,397 @@
...
+
+Si su utility function es una función normal con `def`, se llamará directamente (tal cual la escribes en su código), no en un conjunto de hilos, si la función se crea con` async def`, entonces debes usar await en  esa función cuando la llamas en tu código.
```
Comment (mariacamilagl) [inline docs/es/docs/async.md:389]:
```suggestion
Si tu utility function es una función normal con `def`, se llamará directamente (tal cual la escribes en tu código), no en un threadpool, si la función se crea con `async def`, entonces debes usar await en esa función cuando la llamas en tu código.
```

**Review by mariacamilagl — CHANGES_REQUESTED** [review]
Thank you so much for your effort @alvaropernas 🚀 👍 

This was a long, complex one!

I know the changes may look a bit overwhelming, but they are minor details. Here's the summary of the suggestions: 

1. Removing double spaces and spaces after commas
2. Maintaining consistency between "tu" and "usted"
3. Keeping the central FastAPI terms like "path operation function" in English for consistency
4. Other typos}

Thanks again! 🎉 🍰

File: docs/es/docs/async.md [hunk 26]
```diff
@@ -0,0 +1,397 @@
...
+
+Mientras esperas, vas con tu pareja y eliges una mesa, os sentáis y habláis durante un rato largo  (ya que las hamburguesas son muy impresionantes y llevan un rato prepararse).
+
+Mientras estás  sentado en la mesa con tu  pareja, esperando las hamburguesas, puedes disfrutar ese tiempo admirando lo increíble, guap@ e inteligente que es tu pareja.
+
+Mientras esperas y hablas con tu pareja, de vez en cuando, verificas el número del mostrador para ver si ya es tu turno.
+
+Al final, en algún momento, llega tu turno. Vas al mostrador, coges tus hamburguesas y vuelves a la mesa.
+
+Tú y tu pareja os coméis las hamburguesas y os divertís.
```
[the remark this answers is not in the document]
Comment (alvaropernas) [reply docs/es/docs/async.md:123]: Not sure about this one but... ;)

**Comment by alvaropernas** [comment]
Thanks  Maria, I had doubts about what should be translated.
I will check the extra spaces and the use of "usted" for the next one.

**Comment by mariacamilagl** [comment]
> Thanks Maria, I had doubts about what should be translated.
> I will check the extra spaces and the use of "usted" for the next one.

Hey, no worries. We are also always debating what should or shouldn't be translated. It isn't an exact science at all. 

Thank you for your work! 🍰 🚀

**Review by mariacamilagl — APPROVED** [review]
Thank you! 🚀

**Comment by tiangolo** [comment]
Thanks for your contribution @alvaropernas ! :cake: :rocket: 

And thanks @mariacamilagl for your review and help :nerd_face: :coffee:

**Pull request merged**